4_Zadania_dodatkowe/6.InfiniteSlider/js/app.js

```diff
@@ -0,0 +1,65 @@
+$(document).ready(function () {
+
+    var $prevButton = $('#prevPicture');
+    var $nextButton = $('#nextPicture');
+
+    var $liList = $('li');
+
+    var $ul = $('ul');
+
+    var visibleImageIndex = 0;
+    var imageWidth = $('img').width()+4;   // I tried with outerwidth, css(width) etc but still I have to add 4 because of the space between images; reset of all elements also didn't help
+
+    var $firstLi = $liList.first().clone();
+    var $lastLi = $liList.last().clone();
+    $ul.append($firstLi);
+    $ul.prepend($lastLi);
+
+    $ul.css('right', imageWidth+'px').css('width', imageWidth*8+'px');
```

Comment:
`8` - magical number, it should be calculated from `$liList` (or whatever you name it)